src/main/java/com/eerussianguy/blazemap/feature/mapping/TerrainSlopeLayer.java

```diff
@@ -0,0 +1,56 @@
+package com.eerussianguy.blazemap.feature.mapping;
+
+import com.eerussianguy.blazemap.api.BlazeMapReferences;
+import com.eerussianguy.blazemap.api.builtin.TerrainSlopeMD;
+import com.eerussianguy.blazemap.api.maps.Layer;
+import com.eerussianguy.blazemap.api.maps.TileResolution;
+import com.eerussianguy.blazemap.api.util.ArrayAggregator;
+import com.eerussianguy.blazemap.api.util.IDataSource;
+import com.eerussianguy.blazemap.util.Colors;
+import com.eerussianguy.blazemap.util.Helpers;
+import com.mojang.blaze3d.platform.NativeImage;
+
+public class TerrainSlopeLayer extends Layer {
+
+    public TerrainSlopeLayer() {
+        super(
+            BlazeMapReferences.Layers.TERRAIN_SLOPE,
+            Helpers.translate("blazemap.terrain_slope"),
+            // This should be changed at some point to its own dedicated image
+            Helpers.identifier("textures/map_icons/layer_terrain_isolines.png"),
+
+            BlazeMapReferences.MasterData.TERRAIN_SLOPE
+        );
+    }
+
+    @Override
+    public boolean renderTile(NativeImage tile, TileResolution resolution, IDataSource data, int xGridOffset, int zGridOffset) {
+        TerrainSlopeMD terrain = (TerrainSlopeMD) data.get(BlazeMapReferences.MasterData.TERRAIN_SLOPE);
+
+        foreachPixel(resolution, (x, z) -> {
+            int slope = ArrayAggregator.avg(relevantData(resolution, x, z, terrain.slopemap));
+            paintSlope(tile, x, z, slope);
+        });
+
+        return true;
+    }
+
+    private static void paintSlope(NativeImage tile, int x, int z, int slope) {
+        int shadingRange = 25;
```

Comment:
Yes I just realised I forgot to make this a class variable instead of a function variable. Will fix in the morning.

Reply:
self review... 10/10